### ISSUE TITLE
Useful features with Rails API-only projects

### ISSUE THREAD

**Issue by oyeanuj** [twitter/secure_headers#396]
Hi @oreoshake, had a quick question which I didn't find answered in the docs.

For Rails 4 & 5 projects that are in API-only mode, would you recommend using this project? If so, what configurations/features would be most useful? It seems aimed at classic Rails apps, hence the question.

And maybe we could add this answer to the README, so that no one bothers you again :)

**Comment by oreoshake** [comment]
That's a great suggestion @oyeanuj! I think there will be a lot of opinions on this, namely people will think "if there's no UI what's worth protecting?"

Browsers will do all sorts of things to non-html content. e.g. chrome adds styles to images (:question) and plugins do all sorts of whacky things. I'm of the opinion that at the minimum `default-src 'none'` should be sent. GitHub sends all the headers:

```
Strict-Transport-Security: max-age=31536000; includeSubdomains; preload
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
Content-Security-Policy: default-src 'none'
```

I could see an argument that 

```
X-Frame-Options: deny
X-Content-Type-Options: nosniff
X-XSS-Protection: 1; mode=block
Referrer-Policy: origin-when-cross-origin, strict-origin-when-cross-origin
```

Are unnecessary and wasting extra bytes. Even strict transport security is suspicious because what API client supports it? Nevertheless, people will click on anything and view API responses in browsers all the time. 

In other words, it's a personal decision based on how concerned you are with bandwidth. 

Added a note in https://github.com/twitter/secure_headers/commit/9965178f125a977d1eb248a52003fba5ffa8b321
